sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1576,14 +1576,14 @@ def test_kneighbors_brute_backend(
     metric_params_list = _generate_test_params_for(metric, n_features)
 
     # wminkoski is deprecated in SciPy 1.6.0 and removed in 1.8.0
-    ExceptionToAssert = None
+    warn_context_manager = nullcontext()
     if metric == "wminkowski" and sp_version >= parse_version("1.6.0"):
         if global_dtype == np.float64:
             # Warning from sklearn.metrics._dist_metrics.WMinkowskiDistance
-            ExceptionToAssert = FutureWarning
+            warn_context_manager = pytest.warns(FutureWarning)
         if global_dtype == np.float32:
             # Warning from Scipy
-            ExceptionToAssert = DeprecationWarning
+            warn_context_manager = pytest.warns(DeprecationWarning)
```

Comment:
The CI that uses sets `SKLEARN_RUN_FLOAT32_TESTS=1`, has an older SciPy version so this test is not covered.

Reply:
It should be covered now that we run all jobs in the azure cron job